ipcalc.py

```diff
@@ -562,8 +562,12 @@ def to_tuple(self):
         return (self.dq, self.mask)
 
     def guess_network(self):
-        netmask = 0x100000000 - 2**(32-self.mask)
-        return Network(netmask & self.ip, mask=self.mask)
+        if self.v == 4:
+          netmask = 0x100000000 - 2**(32-self.mask)
+          return Network(netmask & self.ip, mask=self.mask)
+        else:
+          netmask = 2**128 - 2**(128-self.mask)
+          return Network(netmask & self.ip, mask=self.mask)
```

Comment:
Some issues with this:
- 2** is less appropriate than 1<<
- want to reduce some code repetition
- you should be passing on the protocol version; if you don't some IPs will fail

So, perhaps:

```
    if self.v == 4:
        bits = 32
    elif self.v == 6:
        bits = 128
    else:
        raise ValueError('Unexpected version %s' % self.v)
    netmask = ~((1<<(bits - self.mask)) - 1)
    netval = netmask & self.ip
    return ipcalc.Network(netval, mask=self.mask, version=self.v)
```